homeassistant/components/binary_sensor/hive.py

```diff
@@ -0,0 +1,68 @@
+"""
+Support for the Hive devices.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/binary_sensor.hive/
+"""
+import logging
+
+from homeassistant.components.binary_sensor import BinarySensorDevice
+from homeassistant.components.hive import DATA_HIVE
+
+DEPENDENCIES = ['hive']
+
+_LOGGER = logging.getLogger(__name__)
+
+DEVICETYPE_DEVICE_CLASS = {'motionsensor': 'motion',
+                           'contactsensor': 'opening'}
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up Hive sensor devices."""
+    session = hass.data.get(DATA_HIVE)

```

Comment:
I would add a check here before continuing with the setup, to make sure `discovery_info` is not `None`. Add this for all platforms. These platforms should only be setup via discovery.
```py
if discovery_info is None:
    return
session = hass.data.get(DATA_HIVE)
...
```

Reply:
done